autoload/arduino.vim

```diff
@@ -96,6 +100,17 @@ function! arduino#GetArduinoExecutable() abort
   endif
 endfunction
 
+function! arduino#GetBuildPath() abort 
+  if empty(g:arduino_build_path)
+    return ''
+  endif
+  let l:path = g:arduino_build_path
+  let l:path = substitute(l:path, '{file}', expand('%:p'), 'g')
```

Comment:
It doesn't make much sense to me to use the filename of the open file in the path to the build dir. Maybe makes more sense to use the directory that contains the open file?

Reply:
Yeah, actually, I didn't see a use case for it either, but I like to leave the decision to the user (by leaving the information intact).
The directory that contains the open file is already {project_dir}